packages/datetime/src/dateRangeInput.tsx

```diff
@@ -299,30 +302,33 @@ export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, IDat
             >
                 <div className={Classes.CONTROL_GROUP}>
                     <InputGroup
-                        {...startInputProps}
+                        // cast as `any` to` ignore the following error for now:
+                        //   "Property 'ref' of JSX spread attribute is not assignable to target property"
+                        // need to find a better approach here when addressing #943
+                        {...startInputProps as any}
```

Comment:
that sounds pretty reasonable and would have little overall performance impact. either way the perf consideration sounds like a premature optimization. btw that's called _rest_ syntax `{ ref ...startHtmlProps }`